### PR TITLE
[Spells Rework] Quality of Life

### LINKED ISSUE #816: Move syncing
Sometimes, the client doesn't sync the movement of a champion with the server, what makes moving impossible.

This is possibly because of SyncId in movement packet system.

### PR DESCRIPTION
**Pull request by lzardy**
This PR fixes a few issues with movement, adds some scripting API features, and continues with a smooth transition for a full spells rework by shifting some of the file structures related to spells around.

Resolves #816
Progress #757, #566, #527

* Game:
  * Increased tickrate from 30 to 60.
* Packets:
 * Updated LeaguePackets submodule pointer to the latest revision of the master branch and properly accounted for changes in packet names.
  * HandleAutoAttackOption no longer spams chat.
  * EnterVisibilityClient now supports optional usage of TeleportID.
  * EnterVisibilityClient now properly accounts for buffs applied.
  * Convertors changed to PacketExtensions and added automatic creation of MovementData.
  * All packets which used MovementData now use PacketExtensions for automatic creation.
* GameObject:
  * Added Direction as a public variable and removed unneeded Direction variables from GameObject inheriting classes.
  * Added FaceDirection to allow manual setting of direction.
  * Added PlayAnimation (previously in ObjAiBase)
  * SyncId changed to int to remove unneeded casting in packets.
* AttackableUnit:
  * No longer possible to crash from trying to removing a null buff.
  * BuffAddType STACKS_AND_RENEWS now supports removing stacks via RemoveBuff.
  * Removing a buff will now always trigger DeactivateBuff.
  * Added ClearWaypoints function for use in Move function.
  * Move function now uses and assigns to the Direction variable of GameObject.
  * Removed IsMoving function.
  * Added GetParentBuffs for packets.
  * Added SetAnimStates so API is not responsible for sending packets.
* ObjAiBase:
  * OnCollision no longer handles Champion specific interactions.
* LaneMinion:
  * Implemented herd-based pathfinding using QuadTree for WalkToDestination.
* CollisionHandler:
  * QuadDynamic made public and part of the interface (this means we needed to add QuadTree as a NuGet Package to GameServerCore).
* Champions:
  * Collisions with any GameObject other than ObjBuilding and BaseTurret are ignored.
  * TODO: Pathfinding should be responsible for pathing around units so collisions with other units never occur.
* LaneTurret/Nexus/Inhibitor:
  * Now all initialized with their models in SummonersRift.
* Buffs:
  * DeactivateBuff now automatically removes any stat modifiers of the buff.
* Scripting:
  * BuffGameScript OnDeactivate now has a buff and spell parameter.
  * OnUpdate changed to use a float diff.
  * TeleportTo now stops movement automatically.
  * Added FaceDirection.
  * Added GetPointFromUnit for easier skill shot scripting.
  * AddBuff changed to support AttackableUnit.
  * AddParticle/Target changed to support GameObject.
  * Added IsUnitInRange.
  * Added GetClosestUnitInRange for both location and units.
  * Changed DashTo/Target to ForceMovement implementation to be more accurate to League.
  * Added PlayAnimation.
  * Added SetAnimStates.
* Interfaces:
  * Added CastInfo, CastTarget, and SpellSector for Spells.
* General:
  * Added a few more usages of UpdateMoveOrder.
  * PacketNotifier changes fixed positions not syncing for enemies that gain vision of a unit that is moving.
  * Projectile changed to SpellMissile and now inherits GameObject directly.
  * Removed ObjMissile.
  * Moved Projectile Cone, Laser, Spell, and Spell related data to GameObjects/Spell and/or /Missile (as well as namespace).
  * Moved Buff to GameObjects folder (as well as namespace)
  *

### CODE REVIEW

File: GameServerLib/API/ApiFunctionManager.cs [hunk 1]
```diff
@@ -357,6 +392,40 @@ public static List<IAttackableUnit> GetUnitsInRange(Vector2 targetPos, float ran
...
+        /// Acquires the closest alive or dead AttackableUnit within the specified range of another unit.
+        /// </summary>
+        /// <param name="target">Origin of the range to check.</param>
+        /// <param name="range">Range to check from the target position.</param>
+        /// <param name="isAlive">Whether or not to return alive AttackableUnits.</param>
+        /// <returns>Closest AttackableUnit.</returns>
+        public static IAttackableUnit GetClosestUnitInRange(IAttackableUnit target, float range, bool isAlive)
+        {
+            var units = _game.ObjectManager.GetUnitsInRange(target.Position, range, isAlive);
+            var orderedUnits = units.OrderBy(unit => Vector2.DistanceSquared(target.Position, unit.Position));
```
Comment (danil179) [inline GameServerLib/API/ApiFunctionManager.cs:420]: Here you can call `GetClosestUnitInRange(Vector2 targetPos, float range, bool isAlive)`

**Review by danil179 — COMMENTED** [review]
Seems fine. I'll test it and then approve if works.

File: GameServerLib/API/ApiFunctionManager.cs [hunk 2]
```diff
@@ -416,14 +416,7 @@ public static IAttackableUnit GetClosestUnitInRange(Vector2 targetPos, float ran
...
         {
-            var units = _game.ObjectManager.GetUnitsInRange(target.Position, range, isAlive);
-            var orderedUnits = units.OrderBy(unit => Vector2.DistanceSquared(target.Position, unit.Position));
-            if (orderedUnits.First() == target)
-            {
-                return orderedUnits.ElementAt(1);
-            }
-
-            return orderedUnits.First();
+            return GetClosestUnitInRange(target.Position, range, isAlive);
```
Comment (danil179) [inline GameServerLib/API/ApiFunctionManager.cs]: Sorry if I wasn't clear. Just the 2 first lines can be replaced. You still have to check that you don't retrieve target. (at least this is the `if` that happens here)

**Review by danil179 — COMMENTED** [review]
Clarification.

File: GameServerLib/API/ApiFunctionManager.cs (hunk 2 above)
Comment (lzardy) [reply GameServerLib/API/ApiFunctionManager.cs]: I will revert. This function requires the units list in order to get the next closest unit in range (assuming the closest = target).

**Review by danil179 — APPROVED** [review]
Approved. Tested and minion pathing works as expected.

**Pull request merged**